src/script/components/ParticipantAvatar.tsx

```diff
@@ -86,69 +86,61 @@ const ParticipantAvatar: React.FunctionComponent<ParticipantAvatarProps> = ({
   participant,
   ...props
 }) => {
-  const isUser = participant instanceof User && !participant.isService && !participant.isTemporaryGuest();
-  const isService = participant instanceof ServiceEntity || participant.isService;
-  const isTemporaryGuest = !isService && participant.isTemporaryGuest();
-
-  const avatarState = (() => {
-    switch (true) {
-      case isService:
-        return STATE.NONE;
-      case participant.isMe:
-        return STATE.SELF;
-      case participant.isTeamMember():
-        return STATE.NONE;
-      case participant.isBlocked():
-        return STATE.BLOCKED;
-      case participant.isRequest():
-        return STATE.PENDING;
-      case participant.isIgnored():
-        return STATE.IGNORED;
-      case participant.isCanceled() || participant.isUnknown():
-        return STATE.UNKNOWN;
-      default:
-        return STATE.NONE;
-    }
-  })();
+  const isTemporaryGuest = participant instanceof User && participant.isTemporaryGuest();
 
   const clickHandler = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
     onAvatarClick?.(participant, event.currentTarget.parentNode);
   };
 
-  if (isUser) {
+  if (participant instanceof ServiceEntity) {
     return (
-      <UserAvatar
+      <ServiceAvatar
         assetRepository={assetRepository}
         avatarSize={avatarSize}
-        noBadge={noBadge}
-        noFilter={noFilter}
+        participant={participant as ServiceEntity}
         onClick={clickHandler}
-        participant={participant}
-        state={avatarState}
         {...props}
       />
     );
   }
+  let avatarState = STATE.NONE;
+
+  if (participant.isMe) {
+    avatarState = STATE.SELF;
+  } else if (participant.isTeamMember()) {
+    avatarState = STATE.NONE;
+  } else if (participant.isBlocked()) {
+    avatarState = STATE.BLOCKED;
+  } else if (participant.isRequest()) {
+    avatarState = STATE.PENDING;
+  } else if (participant.isIgnored()) {
+    avatarState = STATE.IGNORED;
+  } else if (participant.isCanceled() || participant.isUnknown()) {
+    avatarState = STATE.UNKNOWN;
+  }
 
   if (isTemporaryGuest) {
     return (
       <TemporaryGuestAvatar
         avatarSize={avatarSize}
         noBadge={noBadge}
         onClick={clickHandler}
-        participant={participant}
+        participant={participant as User}
```

Comment:
```suggestion
        participant={participant}
```

Participant should be known as `User` to TypeScript here.